src/mxarray.jl

```diff
@@ -157,13 +158,16 @@ const _mx_is_complex = mxfunc(:mxIsComplex)
 const _mx_is_sparse  = mxfunc(:mxIsSparse)
 const _mx_is_empty   = mxfunc(:mxIsEmpty)
 const _mx_is_struct  = mxfunc(:mxIsStruct)
-const _mx_is_cell    = mxfunc(:mxIsCell) 
+const _mx_is_cell    = mxfunc(:mxIsCell)
 
 
 # getting simple attributes
 
 macro mxget_attr(fun, ret)
-    :( ccall($(fun)::Ptr{Void}, $(ret), (Ptr{Void},), mx.ptr) )
+    quote
+        @assert mx.ptr::Ptr{Void} != C_NULL
```

Comment:
Read the doc at http://julia.readthedocs.io/en/latest/manual/calling-c-and-fortran-code/#auto-conversion

Example of definition at https://github.com/yuyichao/LibALPM.jl/blob/c5b34d781328e64a31f26bbb249563f1cbf8428a/src/handle.jl#L178 and its use at https://github.com/yuyichao/LibALPM.jl/blob/c5b34d781328e64a31f26bbb249563f1cbf8428a/src/handle.jl#L186